waku/v2/node/storage/peer/waku_peer_storage.nim

```diff
@@ -107,7 +117,7 @@ method getAll*(db: WakuPeerStorage, onData: peer_storage.DataProc): PeerStorageR
   ## Retrieves all peers from storage
   var gotPeers = false
 
-  proc peer(s: ptr sqlite3_stmt) = 
+  proc peer(s: ptr sqlite3_stmt) {.raises: [Defect, LPError, ResultError[ProtoError]].} = 
```

Comment:
Should we use `{.raises.}` for non public procs?

Reply:
I think it isn't a must but in general it is prudent as it helps us. Public functions more about providing clean interface to consumer with clear error paths. At least that's how I understand.

Reply:
In some cases, where non public procs are used by public procs, it becomes necessary to add explicit `{.raises.}`.
In this instance, for example, we can be explicit about what `peer()` raises and then catch and handle the exceptions [here](https://github.com/status-im/nim-waku/blob/37c5a6eb222921375a2f9bcedc9133b2c4b1b705/waku/v2/node/storage/peer/waku_peer_storage.nim#L139-L142). This leaves the public proc free of exceptions.

Reply:
I see, I could not find any exact item on the track1 checklist for this one, good to know! makes sense, I will consider it while refactoring other modules.